internal/handlers/common/update.go

```diff
@@ -236,6 +244,63 @@ func processPopFieldExpression(doc *types.Document, update *types.Document) (boo
 	return changed, nil
 }
 
+// processRenameFieldExpression changes document according to $rename operator.
+// If the document was changed it returns true.
+func processRenameFieldExpression(doc *types.Document, update *types.Document) (bool, error) {
+	renameExpression := update.SortFieldsByKey()
+
+	var changed bool
+
+	for _, key := range renameExpression.Keys() {
+		renameRawValue := must.NotFail(renameExpression.Get(key))
+
+		if key == "" || renameRawValue == "" {
+			return changed, NewWriteErrorMsg(ErrEmptyName, "An empty update path is not valid.")
+		}
+
+		// this is covered in validateRenameExpression
+		renameValue := renameRawValue.(string)
+
+		if key == renameRawValue {
+			return changed, NewWriteErrorMsg(
+				ErrBadValue,
```

Comment:
What is the test case for it? Is it something like `bson.D{{"v", "v"}}`? Can we add it to tests?

Reply:
It's tested [here](https://github.com/FerretDB/FerretDB/pull/1753/files#diff-02beb83969c6feb8b48d15aebcd299647e0aa2e8b1e5c531e74eb2c4f636ad85R489-R492)

Reply:
Hmm... Hm... Do you have an idea why codecov didn't catch it? Personally, at first sight, I don't see it.

Reply:
My first guess it's because of the fact that we compare any value with a string (maybe even there's a bug here!!) I'll debug if this if statement works correctly and let you know